intermediate_source/coding_ddpg.py

```diff
@@ -0,0 +1,1229 @@
+# -*- coding: utf-8 -*-
+"""
+TorchRL objectives: Coding a DDPG loss
+======================================
+**Author**: `Vincent Moens <https://github.com/vmoens>`_
+
+"""
+
+##############################################################################
+# Overview
+# --------
+#
+# TorchRL separates the training of RL algorithms in various pieces that will be
+# assembled in your training script: the environment, the data collection and
+# storage, the model and finally the loss function.
+#
+# TorchRL losses (or "objectives") are stateful objects that contain the
+# trainable parameters (policy and value models).
+# This tutorial will guide you through the steps to code a loss from the ground up
+# using TorchRL.
+#
+# To this aim, we will be focusing on DDPG, which is a relatively straightforward
+# algorithm to code.
+# `Deep Deterministic Policy Gradient <https://arxiv.org/abs/1509.02971>`_ (DDPG)
+# is a simple continuous control algorithm. It consists in learning a
+# parametric value function for an action-observation pair, and
+# then learning a policy that outputs actions that maximize this value
+# function given a certain observation.
+#
+# What you will learn:
+#
+# - how to write a loss module and customize its value estimator;
+# - how to build an environment in TorchRL, including transforms
+#   (for example, data normalization) and parallel execution;
+# - how to design a policy and value network;
+# - how to collect data from your environment efficiently and store them
+#   in a replay buffer;
+# - how to store trajectories (and not transitions) in your replay buffer);
+# - how to evaluate your model.
+#
+# Prerequisites
+# ~~~~~~~~~~~~~
+#
+# This tutorial assumes that you have completed the
+# `PPO tutorial <reinforcement_ppo.html>`_ which gives
+# an overview of the TorchRL components and dependencies, such as
+# :class:`tensordict.TensorDict` and :class:`tensordict.nn.TensorDictModules`,
+# although it should be
+# sufficiently transparent to be understood without a deep understanding of
+# these classes.
+#
+# .. note::
+#   We do not aim at giving a SOTA implementation of the algorithm, but rather
+#   to provide a high-level illustration of TorchRL's loss implementations
+#   and the library features that are to be used in the context of
+#   this algorithm.
+#
+# Imports and setup
+# -----------------
+#
+
+import torchrl
+
+# sphinx_gallery_start_ignore
+import warnings
+from typing import Tuple
+
+warnings.filterwarnings("ignore")
+# sphinx_gallery_end_ignore
+
+import torch.cuda
+import tqdm
+
+import torch.multiprocessing
+
+###############################################################################
+# We will execute the policy on CUDA if available
+device = (
+    torch.device("cpu") if torch.cuda.device_count() == 0 else torch.device("cuda:0")
+)
```

Comment:
@svekars this will solve the issue, but it will be slightly slower (not noticeable in this example but a full training loop will be impatcted).
How can we comment this to say "turn on the cuda device for collector for faster training" or smth like it, to let users know that this is just for the sake of fixing the CI?

```suggestion
device = (
    torch.device("cpu") if torch.cuda.device_count() == 0 else torch.device("cuda:0")
)
collector_device = torch.device("cpu")
```

Reply:
Can just add a comment like this:
```
collector_device = torch.device("cpu") # Change the device to ``cuda`` to use CUDA
```